### ISSUE TITLE
parallel = "SOCK" on windows keeps opening new connections

### ISSUE THREAD

**Issue by Ax3man** [TGuillerme/mulTree#32]
I'm running an analysis, basically equivalent to the man page example with:

```
mulTree(mulTree_data, formula = my_formula, parameters = mcmc_parameters,
     priors = mcmc_priors, output = "brain", ESS = 1000, parallel = "SOCK")
```

It seems that this results in two new threads being created whenever the analysis arrives at a new tree, without ever closing the existing connections. This quickly leads to many worked nodes and memory issues.

It seems to me that in the [main loop](https://github.com/TGuillerme/mulTree/blob/master/R/mulTree.R#L195) every iteration `snow::makeCluster` is called, but only [after](https://github.com/TGuillerme/mulTree/blob/master/R/mulTree.R#L273) the whole loop is done, `snow::stopCluster` is called.

I think a fix would involve only calling `makeCluster` once, before the loop. I can try and prepare a PR if you'd like.

**Comment by TGuillerme** [comment]
Hi Wouter,

I've fixed it in the new version `1.2.5`. This version is not released yet but it can be installed by using `install_github("TGuillerme/mulTree", ref = "master")`. Thanks a lot for raising this issue!

Cheers,

Thomas.

**Comment by Ax3man** [comment]
That's some impressive turnaround time! 

Thanks a lot,
W.
